### PR TITLE
Buildkite pipeline to make the zip and upload to s3

### PR DESCRIPTION
**Pull request by ThomasJamesCrawford**

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    concurrency: 1
+    command: .buildkite/deploy.sh heaws $BUILDKITE_BUILD_NUMBER
+    concurrency_group: $BUILDKITE_PIPELINE_SLUG/deploy-heaws
+    depends_on:
+      - build
+      - deploy_test
+    key: deploy_prod
+    plugins:
+      - cultureamp/aws-assume-role#v0.1.0:
+          role: arn:aws:iam::530261158904:role/buildkite-$BUILDKITE_PIPELINE_SLUG
```
Comment (jim-barber-he) [inline .buildkite/pipeline.yml]: I'd imagine it needs to assume the role in production for this.
```suggestion
          role: arn:aws:iam::340978087534:role/buildkite-$BUILDKITE_PIPELINE_SLUG
```

**Review by jim-barber-he — CHANGES_REQUESTED** [review]

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+env:
+  DOCKER_BUILD_TAG: bk-$BUILDKITE_BRANCH-$BUILDKITE_BUILD_NUMBER
+  ECR_REPO_URL: 340978087534.dkr.ecr.ap-southeast-2.amazonaws.com/$BUILDKITE_PIPELINE_SLUG
+
+docker-env: &docker-env
+  environment:
+    - AWS_ACCESS_KEY_ID
+    - AWS_SECRET_ACCESS_KEY
+    - AWS_SECURITY_TOKEN
+    - AWS_SESSION_TOKEN
```
Comment (jim-barber-he) [inline .buildkite/pipeline.yml]: You've defined this, but you haven't used it anywhere below.

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+common-deploy-options: &common-deploy-options
+  agents:
+    docker: 'true'
+  branches: master
+  concurrency: 1
```
Comment (jim-barber-he) [inline .buildkite/pipeline.yml]: Again, you haven't referred to this anywhere below.

**Review by jim-barber-he — APPROVED** [review]
Also another way you can do artifacts uploads and downloads is like how it is done in megatron.
One step does:
```
      - artifacts#v1.3.0:
          upload:
            - build/lambda.zip
```
The other other steps do:
```
      - artifacts#v1.3.0:
          download:
            - build/lambda.zip
```
This would replace the call to `artifact_paths` and the `buildkite-agent artifact` script.
You could then put the `aws s3 ...` command directly into the pipeline.

But what you have is fine.

**Comment by ThomasJamesCrawford** [comment]
> Also another way you can do artifacts uploads and downloads is like how it is done in megatron.
> One step does:
> 
> ```
>       - artifacts#v1.3.0:
>           upload:
>             - build/lambda.zip
> ```
> 
> The other other steps do:
> 
> ```
>       - artifacts#v1.3.0:
>           download:
>             - build/lambda.zip
> ```
> 
> This would replace the call to `artifact_paths` and the `buildkite-agent artifact` script.
> You could then put the `aws s3 ...` command directly into the pipeline.
> 
> But what you have is fine.

oh that's cool

**Pull request merged**